### ISSUE TITLE
p5.js editor console with weird behavior on mutable types

### ISSUE THREAD

**Issue by ScriptHosT12** [processing/p5.js#3700]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [x] Other (editor)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: 0.7.3
- Web browser and version: Chrome 73.0.3683.103
- Operating System: Windows 7 
- Steps to reproduce this:
Create and run this code
```
function a(){
  var v = []
  console.log(v);
  v.push(1);
  console.log(v);
  v.pop();
}
```
On the chrome dev tools it logs `[]` and `[1]`, but on p5.js console it logs twice a blank array

**Comment by outofambit** [comment]
Hi @ScriptHosT12! did you run this code in a sketch on https://editor.p5js.org? or another context? and was it run in `loop` or `setup` or another part of your code?

**Comment by ScriptHosT12** [comment]
Yes it was on a sketch on the editor, about running on loop or on setup it can happen on all events, including draw event

**Comment by limzykenneth** [comment]
It is showing the last known value of the variable at the time of printing the logged value. If you look at the Chrome dev tool result and expand the entry that is `[1]`, you will see that it now says the length is 0. This is likely related more to the editor itself and how it caches console messages and the timing of it.
